### ISSUE TITLE
LastAccessTimeBasedTracking is not supported for the account

### ISSUE THREAD

**Issue by sschmeck** [terraform-providers/terraform-provider-azurerm#11853]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.14.10
hashicorp/azurerm v2.57.0
```

### Affected Resource(s)

* `azurerm_storage_account`

### Terraform Configuration Files

```hcl
terraform {
  required_version = "0.14.10"
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=2.57.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "rg" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "sa" {
  name                      = "examplesa"
  resource_group_name       = azurerm_resource_group.rg.name
  location                  = azurerm_resource_group.rg.location
  account_tier              = "Standard"
  account_kind              = "StorageV2"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true
  is_hns_enabled            = true
  min_tls_version           = "TLS1_2"

  blob_properties {
    cors_rule {
      allowed_headers    = ["*"]
      allowed_methods    = ["GET"]
      allowed_origins    = ["*"]
      exposed_headers    = ["*"]
      max_age_in_seconds = "600"
    }
  }
}
```

### Expected Behaviour

```
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

### Actual Behaviour

```
Error: Error updating Azure Storage Account `blob_properties` "examplesa": storage.BlobServicesClient#SetServiceProperties: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status
=<nil> Code="FeatureNotSupportedForAccount" Message="LastAccessTimeBasedTracking is not supported for the account."

  on main.tf line 24, in resource "azurerm_storage_account" "sa":
  24: resource "azurerm_storage_account" "sa" {
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

We used that setup for about 3 weeks but since Monday (May 24, 2021), the error above occurs. To reproduce the problem you need to setup `is_hns_enabled = true` together with the `blob_properties` block.

The problem occurs only with provider version `>=2.57.0` and seems related to the PR #11301.

### References

* #11301
* #11772

**Comment by TPPWC** [comment]
We face the same problem. We are working with Modules:

`dynamic "delete_retention_policy" {
      for_each = var.blob_delete_retention_policy != 0 ? [1] : []
      content {
        days = var.blob_delete_retention_policy
      }
    }
`
When we delete this part of the module the error is gone.

kr

tom

**Comment by yriveiro** [comment]
> We face the same problem. We are working with Modules:
> 
> ```
> dynamic "delete_retention_policy" {
>       for_each = var.blob_delete_retention_policy != 0 ? [1] : []
>       content {
>         days = var.blob_delete_retention_policy
>       }
>     }
> ```
> 
> We are deploying to west europe.
> 
> kr
> 
> tom

Same configuration, same version, same bug.

**Comment by sschmeck** [comment]
Fixed with version `2.62.0`. Thanks, @yupwei68.
